lib/foreman_ansible_core/playbook_runner.rb

```diff
@@ -11,6 +13,7 @@ class PlaybookRunner < ForemanTasksCore::Runner::CommandRunner
     def initialize(inventory, playbook, options = {})
       super
       @inventory = inventory
+      add_to_known_hosts unless found_in_known_hosts?
```

Comment:
I guess, this would be the line, where you could have an option to opt out the tofu policy (trust on first use).
In that case one were responsible to establish that trust by other means, of course.

Reply:
I've been thinking about this since this comment. Adding a setting here would be tricky as this code is completely decoupled from smart_proxy_ansible (in fact it's only in this repo now because of packaging issues[1] ).        

I think it would make sense to allow people to opt-out, but it's not unreasonable (in fact it's more reasonable than AWX policy, I know it's not a good argument but still..). Can we do this on a separate commit?

[1] - non-SCL package foreman-proxy cannot depend on SCL package tfm-rubygem-foreman_ansible_core

Reply:
Let's first get the rest of the change in then.